cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -596,6 +598,54 @@ func (ht *hcsTask) DumpGuestStacks(ctx context.Context) string {
 	return ""
 }
 
+func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error {
+	if ht.host == nil {
+		return errors.New("task is not isolated")
+	}
+	guestReq := guestrequest.GuestRequest{
+		ResourceType: guestrequest.ResourceTypeMappedDirectory,
+		RequestType:  requesttype.Add,
+	}
+	// If the task is WCOW then we'll be mounting the directory/file with VSMB
+	// and returning.
+	if ht.isWCOW {
+		options := &hcsschema.VirtualSmbShareOptions{
+			NoDirectmap: ht.host.DevicesPhysicallyBacked(),
+		}
+		if req.ReadOnly {
+			options.ReadOnly = true
+			options.CacheIo = true
+			options.ShareRead = true
+			options.ForceLevelIIOplocks = true
+		}
+		guestReq.Settings = &hcsschema.MappedDirectory{
+			ContainerPath: req.UvmPath,
+			HostPathType:  hcsschema.VirtualSmbShareName,
+			ReadOnly:      req.ReadOnly,
+		}
+		_, err := ht.host.AddVSMB(ctx, req.HostPath, guestReq, options)
+		return err
+	}
+
+	// The task is LCOW so we'll be mounting the directory/file with Plan9.
+	st, err := os.Stat(req.HostPath)
```

Comment:
Why wouldn't we `Stat` in the WCOW case?

Reply:
AddVSMB does a stat internally

Reply:
I was wondering if we should move the single file mapping logic inside of AddPlan9 instead of having the caller supply allowedNames. But you're not limited to a single file with the current model, you can pass in however many files you want so I'm torn on that.

Reply:
I think it's fine to leave the restrict file as is for now. However, I think it's weird to have this different expectation between callers of adding VSMB vs 9P. What happens if you pass a path that doesn't exist for a 9P share? Can we make that function the same as WCOW?

Reply:
You're referring to doing a stat in AddPlan9 to keep it consistent with VSMB correct? I can add that if so. How would you like the callers supplying allowedNames to function however? Right now they just do stat calls like above externally to decide whether to set restrict to true and populate allowedNames.

Reply:
Okay, you're right that without changing how `AddPlan9` is called, we probably need to keep doing this stat here. On second thought I think that's okay for now. We can come up with a better interface later if needed.

Reply:
I don't think there are any callers of `AddPlan9` which actually use the `allowedNamed` array to pass multiple file names as of now (correct me if I am wrong). So I think we can change this interface right now to make both `AddVSMB` and `AddPlan9` have the same interface. 
In fact this makes me wonder how `AddPlan9` will work if we have multiple files from different directories in the `allowedNames` array? Because as per my understanding we can not add multiple files from different directories in the same `AddVSMB` call.